libkbfs/bcache.go

```diff
@@ -242,6 +243,18 @@ func (b *BlockCacheStandard) makeRoomForSize(size uint64, lifetime BlockCacheLif
 func (b *BlockCacheStandard) PutWithPrefetch(
 	ptr BlockPointer, tlf tlf.ID, block Block, lifetime BlockCacheLifetime,
 	hasPrefetched bool) (err error) {
+	// Just in case we tried to cache a block type that shouldn't be cached,
+	// return an error. This is an insurance check. That said, this got rid of
+	// a flake in TestSBSConflicts, so we should still look for the underlying
+	// error.
+	switch block.(type) {
+	case *DirBlock:
+	case *FileBlock:
+	case *CommonBlock:
+		return errors.New("attempted to Put a common block")
```

Comment:
Should we only return this error when `lifetime == NoCacheEntry`?  Or should we get rid of `NoCacheEntry` altogether if we want to force the caller into policing this?